### PR TITLE
chore: Add hook support to contract tests

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: contract-tests/hook.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  #
+  # @param hook_context [LaunchDarkly::Interfaces::Hooks::EvaluationContext]
+  # @param data [Hash]
+  #
+  def before_evaluation(hook_context, data)
+    payload = {
+      evaluationHookContext: {
+        flagKey: hook_context.key,
+        context: @context_filter.filter(hook_context.context),
```
Comment (cwaldren-ld) [inline contract-tests/hook.rb:30]: Why the filtered context?

**Comment by cwaldren-ld** [comment]
`evaluationHookContext` -> `evaluationSeriesContext`
`evaluationHookData` -> `evaluationSeriesData` 

(latest updates to the contract test PR)

File: contract-tests/hook.rb (hunk 1 above)
Comment (keelerm84) [reply contract-tests/hook.rb]: The SDK has no other way of converting a context into a hash other than attribute redaction. We haven't needed it yet, and unless the tests require some special handling of private attributes (which seems odd), I figured I could get away with this.

**Comment by keelerm84** [comment]
> `evaluationHookContext` -> `evaluationSeriesContext` `evaluationHookData` -> `evaluationSeriesData`
> 
> (latest updates to the contract test PR)

Thanks. Updated!

**Review by cwaldren-ld — APPROVED** [review]

**Pull request merged**